### ISSUE TITLE
The deleted messages remain on the node for 14 days

### ISSUE THREAD

**Issue by jimgillo** [oxen-io/session-android#535]
An user found this :
**why can i see deleted messages after reinstalling.**

Developper answer this:
**encrypted messages are stored on the Service Node network inside your swarm until the TTL expiry which is 14 days, message deletion only occurs locally, global deletion occurs when messages expire from your swarm**

**This is not an acceptable solution for a secure messenger.Deleting messages should send a message to the Node to delete pending messages.**

You must always stand on the other side: you are in a Chinese police station, tortured, you give your password thinking that you have erased everything and unfortunately when the Chinese policeman opens your phone he receives all the messages waiting on the Node for 14 days (or after deleting the app and reinstalling). Conclusion: he has all the evidence that you are an enemy of the people for journalism and you are executed the next morning.

**Comment by nielsandriesse** [comment]
Hi @jimgillo, though I agree that Session should clear your messages from the Service Node network when you clear your data, I'd like to make a few points:

• The scenario you've described can only occur if you've written down your recovery phrase somewhere, or if you know it by heart. If you didn't there's no way to restore your private key, and therefore there's no way to retrieve your messages and decrypt them even if they're still on the network.
• Implementing this feature is on our near-term roadmap, but we will need Storage Server support before we can implement it.
• Please refrain from using such extreme examples. We understand the importance of the issue you're raising even without it.

**Comment by nielsandriesse** [comment]
I've removed the example you gave from the issue, keeping all the relevant information.

**Comment by noghx** [comment]
The example was about police force you to give your seed... yeah in some countrys they even use physical violence for this.

Now imagine u give your seed and they RESTORE YOUR SELFDESTRUCTED MESSAGES.

**Comment by nielsandriesse** [comment]
Like I mentioned, we're aware of the issue and we're working on it. I also understand the example you gave perfectly well. The reason I removed it is because it was unnecessary and very extreme. We do have rules here that you're required to follow. Also, don't upvote your own comments next time. It doesn't help you :)

**Comment by noghx** [comment]
"Also, don't upvote your own comments next time. It doesn't help you :)"

The question is: Do i need help or do you need help to close 4 month old critical security holes?

**Comment by jimgillo** [comment]
I did not vote for my own comment, you are wrong. In addition, Keejeef has found the solution: allow the user to set the TTL he wants for his messages (by using the message user expiration time).

**Comment by chairmantheboring** [comment]
> allow the user to set the TTL


@jimgillo @noghx This would make you stand out, because the service node network needs to know the TTL. Not only that, but it would allow an adversary controlling a portion of the service nodes (if they got lucky, and you got unlucky) to potentially know which Session IDs you are messaging. Session does a great job at restoring your identity from the seed (most times). If you are afraid of an adversary compromising your device in the future, then uninstalling the application when you are not using it and only having it installed when necessary is the way to go.


I'm interested to see what the developers come up with as a solution to this, but the reality is that any way to make messages exist shorter (without changing these settings for everyone)/be deleted early will make your identity stand out to a malicious service node.

**Comment by IERINMEX** [comment]
Duplicate of #407

**Comment by IERINMEX** [comment]
I think a successful download of a self destructing message should trigger a delete on the service node. When i can find time I'll try to write some code for this (unless someone gets around to it before me). custom ttl doesn't seem like the best solution in my opinion.

**Comment by nielsandriesse** [comment]
The proposed solution of setting the TTL based on the message expiration setting is problematic. Messages expire locally **from the moment they're received**, so that's what we want to match on the storage server. If we set the TTL to the disappearing messages setting when sending a message, we're expiring messages on the storage server relative to when they were **sent**. Imagine a thread with disappearing messages set to 1 minute. The messages would be almost guaranteed to expire on the storage server before the other party has received them.

@noghx we're doing what we can. Also, as I mentioned before, this feature is dependent on the storage server (i.e. we can't do it until the network supports this). If you want this to happen faster the best way to make that happen is to be supportive of the team and the project.

**Comment by IERINMEX** [comment]
> The proposed solution of setting the TTL based on the message expiration setting is problematic. Messages expire locally **from the moment they're received**, so that's what we want to match on the storage server. If we set the TTL to the disappearing messages setting when sending a message, we're expiring messages on the storage server relative to when they were **sent**. Imagine a thread with disappearing messages set to 1 minute. The messages would be almost guaranteed to expire on the storage server before the other party has received them.
> 
> @noghx we're doing what we can. Also, as I mentioned before, this feature is dependent on the storage server (i.e. we can't do it until the network supports this). If you want this to happen faster the best way to make that happen is to be supportive of the team and the project.

Could a successful download by a client send a receipt of sorts to the server that would trigger deletion on the server node/s it's stored on?

**Comment by nielsandriesse** [comment]
> Could a successful download by a client send a receipt of sorts to the server that would trigger deletion on the server node/s it's stored on?

That's more or less the plan!

**Comment by nielsandriesse** [comment]
This problem will partly be fixed as of the upcoming round of releases. When you clear your account you'll now be given the option to also delete all your data on the network. The reason this took a while is because we needed to do a Storage Server upgrade before we could do this. The other parts of this issue will be addressed in the near future.

**Comment by eyeberic** [comment]
Bumping on this Issue as this is still present with current desktop (1.6.11) and Android (1.11.8) clients. 

Messages that self-destructed based on disappearing message timer can still be recovered during account recovery.

**Comment by noghx** [comment]
So how long it will take to solve this problem?

**Comment by noghx** [comment]
Thanks for your fast response :)

**Comment by pixloptr** [comment]
> > Could a successful download by a client send a receipt of sorts to the server that would trigger deletion on the server node/s it's stored on?
> 
> That's more or less the plan!

Really looking forward to this being implemented 😊

**Comment by noghx** [comment]
Something new?

**Comment by guest346** [comment]
As far as I understand, there is no solution to the problem. Is there any news for a fix?

Make it compulsory to create a password. Without knowing it, a person will not be able to connect a second device or reinstall the program. You get a HUGE security hole. I am very surprised that this has not been fixed until now. These things need to be fixed in the first place.
